stablehlo/reference/Tensor.h

```diff
@@ -99,6 +109,11 @@ class Tensor {
 
  private:
   llvm::IntrusiveRefCntPtr<detail::Buffer> impl_;
+  std::vector<int64_t> strides_;
+
+  // Flattens multi-dimensional index `indices` of a tensor to a linearized
+  // index into the underlying storage using the dimension strides `strides_`.
+  int64_t flattenIndices(const std::vector<int64_t> &indices) const;
```

Comment:
Can we turn this function into a free function in Tensor.cpp? If there's no strong reason for it to be a member function, it shouldn't be.

Reply:
We keep strides are part of Tensor object as recomputing it every-time we call Tensor::get/Tensor::set seems redundant. With `stride`, `get`, and `set` being part of Tensor class, it makes sense to me for  `flattenIndices` to be a (private) member function as well  as it uses the `strides` to provide index to the internal implementation of Tensor: Any change in the internal implementation might need changes in the function because of its close connection with how a tensor is internally stored.   
WDYT?

Reply:
not appllicable.